### PR TITLE
JITMS: adding new JITM styles

### PR DESCRIPTION
**Pull request by jeffgolenski**
Adding 2 new JITM styles and cleaning up the original JITM

To do:
- [x] clean up woo commerce JITM (visually)
- [ ] clean up original JITM styles. Update outdated css
- [x] add styles for new JITM banners (calypso styled)

**Current JITM style example:**
old-jitm-style

**New JITM style examples:**
screen shot 2017-05-15 at 4 59 26 pm

* Can change border color if desired.
* Can have title + description or just a title
* Can have gray or primary blue CTA button
* JITMs will dynamically resize/realign based on the content. In order words, If you only have a title in the JITM, it'll be vertically centered, and when you add a description, it'll automatically ensure both are vertically centered. ;-)
*Work well on mobile

Important to note that the CTA buttons have a max width in order to be easier to predict and control how they'll be displayed on various screen sizes—which can bump the button text to two lines if its long. We should try and re-word some of the JITMS so the button is one word and most of the verbs are in the description.

**TO TEST:**

1. Get this branch synced up to your sandbox.
2. Visit `yousite.com/wp-admin/update-core.php` where the sample JITMs sit ** Ensure VaultPress is deactivated to see these particular JITMs
3. Sample woo JITM is hanging out at `yoursite.com/wp-admin/edit.php?post_type=shop_order` when woo commerce is installed.

** If you've dismissed JITMs in the past and no longer see them, you'll need to reset your JITM options. Do so by SSH'ing into your sandbox and clearing them with `cd public_html` and then `wp jetpack options delete hide_jitm`

### CODE REVIEW

**Comment by withinboredom** [comment]
@jeffgolenski don't worry about actually updating the individual markup for each type of JITM. Your examples will work great and I'll just merge this into my new JITM branch.

**Comment by jeffgolenski** [comment]
@withinboredom Sounds good. now that I've cleared out the other priorities from last week I'll be continuing a long with this. I've got a little more code to clean up and some styles to add for a 3rd banner style and then we'll be gold.

**Comment by jeffgolenski** [comment]
@jessefriedman @richardmuscat Can you take a look at the example images above and let me know if you have any questions? They're very flexible and still very subtle.. most importantly, they adapt to the content on the fly (As described above) and are based on the calypso banners. Give me the thumbs up and I'll replace all the old JITM style code to match these. 

Then @withinboredom can do his thing.

**Comment by jessefriedman** [comment]
They look excellent

As for the border color on the left, did you have ideas on what those colors could mean? Or were you just showing examples?

As for the fixed width button even super short cta's can translate to long text, how will these compensate? 

Have you tested them in tight quarters like the left hand side of the media uploader?

**Comment by jeffgolenski** [comment]
@jessefriedman 

> As for the border color on the left, did you have ideas on what those colors could mean? Or were you just showing examples?

My thoughts here were just to bring over the functionality of the calypso banners. If we decide we want to swap the colors we can... but we need to keep in mind that specific colors are related to plan upgrade nudges... so if we're staying away from those type of JITMS I recommend sticking with the green border. 

> As for the fixed width button even super short cta's can translate to long text, how will these compensate?

Long text is fine, provided its one word. If the CTA buttons are multiple words then we can't control line break—and nor should we be mico-managing the design like that... the jitms and buttons within them will still function just fine either way.

example:
screen shot 2017-05-16 at 4 31 40 pm

> Have you tested them in tight quarters like the left hand side of the media uploader?

Still doing some testing. Will report back tomorrow. Just wanted to make sure everyone was okay with the visual direction here.

**Comment by jessefriedman** [comment]
Great Jeff, in that case these are gtg from me. Great job.

**Comment by jeffgolenski** [comment]
Revised JITM for enabling backups after a post is published

screen shot 2017-05-17 at 11 53 35 am
screen shot 2017-05-17 at 11 55 30 am

**Comment by jeffgolenski** [comment]
Revised woo JITM:

old:
screen shot 2017-05-17 at 11 59 00 am

new:
screen shot 2017-05-17 at 1 04 18 pm


cc @kellychoffman

**Comment by jeffgolenski** [comment]
Alright @withinboredom  - I think we're good to go on my end here. All the styling is setup. Once you employ your new method of triggering these JITMs I can do some more testing and cleanup if need be.

**Comment by jeffgolenski** [comment]
@dereksmart I tweaked the format of the JITMS for a few existing jitms. Can you give the code a once-over and make sure I didn't screw up any of the tracks data you put in?

**Comment by jeffstieler** [comment]
@avivapinchas @MarinaWoo the styling has changed on the JITMs, perhaps a copy change is warranted as well? (See https://github.com/Automattic/jetpack/pull/7117#issuecomment-302156590)

The new styling looks to allow for a bolder title with text underneath. What should the WCS JITMs say differently, if anything? Here's a backup one, for reference: https://github.com/Automattic/jetpack/pull/7117#issuecomment-302136036

**Comment by avivapinchas** [comment]
I'd edit to: 

> **New: Show live rates from USPS rates**
> Added bonus: print shipping labels without leaving WooCommerce.

**Comment by jeffstieler** [comment]
@jeffgolenski do you mind if I sneak in a copy change for our JITM here?

**Comment by jeffgolenski** [comment]
@jeffstieler have at it jeffrey!

**Comment by jeffstieler** [comment]
It looks like there's more than styling tweaks here (I'm seeing some different tracking code as well) - I'm going to hold off on making the copy changes until I have more time to grok everything that's been changed in the JITMs. 👍

**Comment by rickybanister** [comment]
@jeffgolenski the `x` icon—is that dashicons? It looks off when compared to Calypso, I suppose we are unable to use gridicons since these exist outside of Jetpack?

**Comment by jeffgolenski** [comment]
@rickybanister 

> I suppose we are unable to use gridicons since these exist outside of Jetpack?

Precisely. Unless we make a huge push to load gridicons everywhere. It's a very similar `x` so I didn't feel that taking a day to do so has high ROI

**Comment by dereksmart** [comment]
Looks great.  Tracks are still sending correctly.  @withinboredom you can incorporate tracks into the following PR with these data attributes in a link element: 

```
<a
data-jptracks-name="nudge_click"
data-jptracks-prop="jitm-vault"
class="jptracks" // necessary class name
>
```

Code that tracks it is [here](https://github.com/Automattic/jetpack/blob/master/_inc/lib/tracks/tracks-ajax.js)

**Comment by withinboredom** [comment]
Implemented in #7058
